transmogrifier/sources/xml/ead.py

```diff
@@ -524,6 +456,123 @@ def get_identifiers(cls, source_record: Tag) -> list[timdex.Identifier] | None:
                 )
         return identifiers or None
 
+    @classmethod
+    def get_languages(cls, source_record: Tag) -> list[str] | None:
+        languages = []
+        collection_description_did = cls._get_collection_description_did(source_record)
+        for langmaterial_element in collection_description_did.find_all(
+            "langmaterial", recursive=False
+        ):
+            languages.extend(
+                [
+                    language
+                    for language_element in langmaterial_element.find_all("language")
+                    if (language := cls.create_string_from_mixed_value(language_element))
+                ]
+            )
```

Comment:
For field methods that involved nested for loops, the following pattern is applied:
* instantiate list instance (`languages = []`)
* external for loop statement
* extend list instance to list comprehension if inner for loop